lnwallet/close_test.go

```diff
@@ -0,0 +1,239 @@
+package lnwallet
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+)
+
+// genValidAmount generates valid bitcoin amounts (non-negative).
+func genValidAmount(t *rapid.T) btcutil.Amount {
+	return btcutil.Amount(
+		rapid.Int64Range(0, 21_000_000*100_000_000).Draw(t, "amount"),
+	)
+}
+
+// genCoopCloseFee generates a reasonable non-zero cooperative close fee.
+func genCoopCloseFee(t *rapid.T) btcutil.Amount {
+	// Generate a fee between 250-10000 sats which is a reasonable range for
+	// closing transactions
+	return btcutil.Amount(
+		rapid.Int64Range(250, 10_000).Draw(t, "coop_close_fee"),
+	)
+}
+
+// genChannelType generates various channel types, ensuring good coverage of
+// different channel configurations including anchor outputs and other features
+func genChannelType(t *rapid.T) channeldb.ChannelType {
```

Comment:
From the rapid go page:
```
Compared to [testing.F.Fuzz](https://pkg.go.dev/testing#F.Fuzz), rapid shines in generating complex
structured data, including state machine tests, but lacks coverage-guided feedback and mutations. Note
that with [MakeFuzz](https://pkg.go.dev/pgregory.net/rapid#MakeFuzz), any rapid test can be used as a 
fuzz target for the standard fuzzer.
```
It would be good to convert all of our rapid tests to fuzz tests. Rapid would be used to generate the initial data (I'm guessing it would be better than go's fuzzing engine for that) and then the fuzzer would mutate off of that.